02-basic-url-design.md

```diff
@@ -0,0 +1,146 @@
+# Basic URL Design
+
+This chapter will show you how to design the 
+REST URL's for a specific resource. 
+
+It will not cover the JSON structure of the
+response and request bodies. This topic is
+covered in chapter: JSON Object Design.
+
+Here is a quick cheat sheet for a 'car' resource:
+
+| URL           | method | response         |
+| ------------- |--------| -----------------|
+| /cars         | GET    | A list of cars   |
+| /cars/1       | GET    | A single car     |
+| /cars/1       | POST   | The created car  |  
+| /cars/1       | PUT    | The updated car  |
+| /cars/1       | PATCH  | The updated car  |
+| /cars/1       | DELETE | The deleted car  |
+
+# Anatomy of a Resource.
+
+A REST API is a collection of resources. A Resource
+represents something in the domain of your application.
+For example: a domain for a webshop might have
+Stock, Product and Customer resources.
+
+A resource is something that can be manipulated.
+For example we can add, remove and update a resource.
+
+A resource can also be queried: we can get a list of
+a particular resource, but we can retrieve the details of
+a specific resource.
+
+Before REST we would create specific URL's for each
+operation or query type. One URL to request a list,
+one to delete a resource etc etc. With REST we define
+one 'URL' for a resource and use verbs to manipulate
+a resource.
+
+So what are these 'verbs'. 
+
+# Request Methods
+HTTP supports multiple 'requests methods' for a single
+URL. Request methods are the 'verbs' which specify what 
+you want to happen when you request that url.
+
+The Request Methods that REST commonly use are: GET, POST, PUT, 
+PATCH and DELETE. These are all specified in the HTTP standard and
```

Comment:
You can link here to https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.3